htdocs/contrat/card.php

```diff
@@ -802,6 +802,8 @@
 			$result = $objectline->update($user);
 			if ($result < 0) {
 				$error++;
+				$action = 'editline';
+				$_GET['rowid'] = GETPOST('elrowid');
```

Comment:
@FHenry : are you sure to set _GET php magic variable ? i will have a look at the other part of the code but that's strange ...

Reply:
ok that's the same code as for example line 698 ...

Reply:
Yes exactly the same code on failure as in 
```
elseif ($action == 'updateline' && $user->hasRight('contrat', 'creer') && !GETPOST('cancel', 'alpha')) {
		$error = 0;
		$predef = '';

		if (!empty($date_start_update) && !empty($date_end_update) && $date_start_update > $date_end_update) {
			setEventMessages($langs->trans("Error").': '.$langs->trans("DateStartPlanned").' > '.$langs->trans("DateEndPlanned"), null, 'errors');
			$action = 'editline';
			$_GET['rowid'] = GETPOST('elrowid');
			$error++;
		}

```

I Think the $_GET['rowid'] affectation is to scroll to line and/or set hidden input
I've just use the same code in case of failed update

Reply:
> ok that's the same code as for example line 698 ...

Yes, it's true that code is strange. It is a bad practice. But it can be accepted in maintenance branch because it is already done this way in samefile few lines before.
This bad coding behaviour has been fixed into the develop branch.